lxd/storage/drivers/driver_btrfs_volumes.go

```diff
@@ -1341,7 +1341,7 @@ func (d *btrfs) migrateVolumeOptimized(vol Volume, conn io.ReadWriteCloser, volS
 
 			_, snapName, _ := shared.InstanceGetParentAndSnapshotName(snap.name)
 
-			if snapName == volSrcArgs.Snapshots[0] {
+			if len(volSrcArgs.Snapshots) > 0 && snapName == volSrcArgs.Snapshots[0] {
```

Comment:
@monstermunchkin does this fix the `Error: Failed instance creation: Error transferring instance data: Got error reading migration source: websocket: close 1000 (normal)` error?

Reply:
Was the target LXD crashing?

Reply:
> does this fix the `Error: Failed instance creation: Error transferring instance data: Got error reading migration source: websocket: close 1000 (normal)` error?

Yes.

> Was the target LXD crashing?

No, only the source.

Reply:
Ah I see thanks.

Reply:
Is there a similar issue with the ZFS optimized refresh mode?

Reply:
> Is there a similar issue with the ZFS optimized refresh mode?

No, there isn't. I just checked.